examples/multigpu_advtrain/model.py

```diff
@@ -26,18 +27,179 @@ def unify_device_name(dname):
     return dname.lower().replace('device:', '')
 
 
+class MLP(Model):
```

Comment:
if we're going to keep MLP and the corresponding layer definitions here, would it make sense to keep it in the tutorials folders and add the new implementation based on `tf.layers` as a new implementation that will eventually replace it? (maybe through some deprecation warning?) 

@goodfeli ?

Reply:
It's only used in Examples now though, that's why I moved it here.
I reworked models into more tensorflow compliant APIs since (1) they do the same thing and (2) it's an additional hurdle to have to learn a mini-layer framework in addition to cleverhans.

Reply:
I think it's fine to move it into the multi-GPU example if that's the only place it's still used. Since it was in the tutorial before, people shouldn't be depending on it. It's not part of the main library API.

Reply:
SGTM